packages/sanity/src/core/studio/packageVersionStatus/PackageVersionStatusProvider.tsx

```diff
@@ -1,93 +1,95 @@
-import {Box, useToast} from '@sanity/ui'
-import {type ReactNode, useCallback, useEffect, useRef} from 'react'
+import {type ReactNode, useCallback, useEffect, useMemo, useRef, useState} from 'react'
+import {PackageVersionInfoContext} from 'sanity/_singletons'
 import semver from 'semver'
 
-import {Button} from '../../../ui-components'
 import {hasSanityPackageInImportMap} from '../../environment/hasSanityPackageInImportMap'
-import {useTranslation} from '../../i18n'
 import {SANITY_VERSION} from '../../version'
-import {checkForLatestVersions} from './checkForLatestVersions'
+import {fetchLatestVersions} from './fetchLatestVersions'
 
-// How often to to check last timestamp. at 30 min, should fetch new version
-const REFRESH_INTERVAL = 1000 * 30 // every 30 seconds
-const SHOW_TOAST_FREQUENCY = 1000 * 60 * 30 // half hour
+// How often to check for new versions
+const POLL_INTERVAL_MS = 1000 * 60 * 30 // check every 30 minutes
+const CHECK_THROTTLE_TIME_MS = 1000 * 10 // prevent checking more often than every 10s
 
-const currentPackageVersions: Record<string, string> = {
-  sanity: SANITY_VERSION,
-}
+const currentPackageVersions = [
```

Comment:
In theory, we could add other packages here in the future, but I think since all we care about currently is the `sanity`-package version, we could consider simplifying this by not supportin an array of versions (makes the code a bit harder to work with)